python/mxnet/gluon/block.py

```diff
@@ -1079,7 +1100,7 @@ def _build_cache(self, *args):
                     param = Parameter(name, dtype=param_data.dtype)
                     param._var_name = name
                     serialization_name = name  # HybridBlock.export
-                    param._load_init(param_data, args[0].context)
+                    param._load_init(param_data, param_data.context)
```

Comment:
I guess args[0] was used because all args must have the same context? With this change, are we allowing multiple contexts to params?

Reply:
before we were assuming that args could only be NDArrays, so there was a context. But here now args can be mx.sym.var which doesnt have a context, so we need to get the context from somewhere else.